src/context/AppReducer.js

```diff
@@ -0,0 +1,21 @@
+import{ theme } from "./theme";
+
+export default function reducer(state, action){ 
+    switch (action.type) {
+        case "SET_SEARCH": {
+            return { 
+                ...state, 
+                search: action.payload
+            }
+        }
+        case "TOGGLE_THEME": {
+            const newThemeKey = state.currentTheme.id === "dark" ? "light" : "dark";
+            return { 
+                ...state,
+                currentTheme: theme[newThemeKey] 
+            };
+        }
+        default: 
+            throw new Error("Unknown Action");
+    }
+}
```

Comment:
Would you mind adding a couple of unit tests for your reducer?. This way you can check that your actions actually mutate the state of your app.